status/discover.go

```diff
@@ -13,32 +14,51 @@ var saToken string
 var clientFactory kubernetes.ClientFactory
 
 func DiscoverJaeger() (url string, err error) {
-	if config.Get().ExternalServices.Jaeger.URL != "" {
-		return config.Get().ExternalServices.Jaeger.URL, nil
-	}
+	conf := config.Get()
 
+	if conf.ExternalServices.Jaeger.URL != "" {
+		log.Debugf("Detected url %s configured for tracing service", conf.ExternalServices.Jaeger.URL)
+		return conf.ExternalServices.Jaeger.URL, nil
+	}
+	log.Debugf("Kiali is looking for Tracing service ...")
+	tracing := conf.ExternalServices.Jaeger.Service
+	jaeger := conf.ExternalServices.Jaeger.JaegerService
+	ns := conf.ExternalServices.Jaeger.Namespace
 	// Check if there is a Tracing service in the namespace
-	url, err = discoverUrlService(config.Get().ExternalServices.Jaeger.ServiceNamespace, config.Get().ExternalServices.Jaeger.Service)
+	url, err = discoverUrlService(ns, tracing)
 	if err != nil || url == "" {
 		// Check if there is a Jaeger-Query service in the namespace
-		url, err = discoverUrlService(config.Get().ExternalServices.Jaeger.ServiceNamespace, config.Get().ExternalServices.Jaeger.JaegerService)
+		url, err = discoverUrlService(ns, jaeger)
+		if err != nil || url == "" {
+			log.Debugf("Services %s or %s not detected  in namespace %s", jaeger, tracing, ns)
+		}
 	}
+	conf.ExternalServices.Jaeger.URL = url
+	config.Set(conf)
 	return url, err
 }
 
 func DiscoverGrafana() (url string, err error) {
+	conf := config.Get()
 	// If display link is disable in Grafana configuration return empty string and avoid discover
-	if !config.Get().ExternalServices.Grafana.DisplayLink {
+	if !conf.ExternalServices.Grafana.DisplayLink {
+		log.Debugf("Grafana display link is disabled in the configuration.")
 		return "", nil
 	}
-	if config.Get().ExternalServices.Grafana.URL != "" {
-		return config.Get().ExternalServices.Grafana.URL, nil
+	if conf.ExternalServices.Grafana.URL != "" {
+		log.Debugf("Detected url %s configured for tracing service", config.Get().ExternalServices.Grafana.URL)
+		return conf.ExternalServices.Grafana.URL, nil
 	}
-	return discoverUrlService(config.Get().ExternalServices.Grafana.ServiceNamespace, config.Get().ExternalServices.Grafana.Service)
+	log.Debugf("Kiali is looking for Grafana service ...")
+	url, err = discoverUrlService(config.Get().ExternalServices.Grafana.ServiceNamespace, config.Get().ExternalServices.Grafana.Service)
+	conf.ExternalServices.Grafana.URL = url
+	config.Set(conf)
+	return url, err
 }
 
 func discoverUrlService(ns string, service string) (url string, err error) {
 	if saToken == "" {
+		log.Debugf("Reading sa Token")
```

Comment:
This debug message will be confusing because just seeing "Reading sa Token" doesn't indicate why/what it is doing this for. I recommend changing to Tracef. Also, perhaps say, "Reading the SA token in order to detect external service".